src/game/game_6_0.c

```diff
@@ -0,0 +1,329 @@
+#include "global.h"
+#include "core.h"
+#include "malloc_vram.h"
+#include "game/game.h"
+#include "game/dust_effect_braking.h"
+#include "game/mp_unknown_task.h"
+#include "game/player_actions.h"
+#include "game/player_callbacks_1.h"
+#include "game/playerfn_cmds.h"
+#include "game/stage/stage.h"
+
+#include "constants/animations.h"
+#include "constants/move_states.h"
+#include "constants/player_transitions.h"
+
+extern u16 gUnknown_080D6736[115][2];
+
+typedef struct {
+    /* 0x00 */ u8 unk0;
+    /* 0x04 */ u32 unk4; // TODO: Check the type!
+} Game_6_0_Task; /* size: 0x8 */
+
+extern void InitNewInputCounters(void);
+void Task_8023FC0(void);
+extern void sub_801FC2C(void);
+void AllocateCharacterStageGfx(Player *, UNK_3005A70 *);
+void AllocateCharacterMidAirGfx(Player *, UNK_3005A70 *);
+void InitializePlayer(Player *);
+void TaskDestructor_802A07C(struct Task *);
+
+s32 gUnused_03005B78 = 0;
+u8 gUnknown_03005B7C = 0;
+
+void sub_80213C0(u32 UNUSED characterId, u32 UNUSED levelId, Player *player)
+{
+#ifndef NON_MATCHING
+    register Player *p asm("r5") = player;
+#else
+    Player *p = player;
+#endif
+
+    s32 unk60 = p->unk60;
+    struct Task *t;
+    Game_6_0_Task *gt;
+    s32 mode;
+
+    t = TaskCreate(Task_8023FC0, sizeof(Game_6_0_Task), 0x3000, 0,
+                   TaskDestructor_802A07C);
+    p->spriteTask = t;
+    gt = TaskGetStructPtr(t);
+    gt->unk0 = unk60;
+    gt->unk4 = 0;
+
+    if (IS_MULTI_PLAYER) {
+        if (gGameMode != GAME_MODE_MULTI_PLAYER_COLLECT_RINGS) {
+            gRingCount = 1;
+        } else {
+            gRingCount = 0;
+        }
+    }
+
+    InitializePlayer(p);
+
+    PLAYERFN_SET(PlayerCB_8025318);
+
+    sub_801F754();
+    sub_801FC2C();
+
+    mode = gInputRecorder.mode;
```

Comment:
Btw idk if `mode` was required for match here, but seems like you should just get a match with

```
if ((gInputRecorder.mode == RECORDER_RECORD)) {
        InputRecorderLoadTape();
        gInputRecorder.mode = RECORDER_RECORD;
}
```
Perhaps not though